### DISCUSSION TITLE
How to generate soil images with different pore size distribution?

### DISCUSSION THREAD

**Discussion by morendong** [PMEAL/porespy#886] [Q&A]
Hi all, 
I tried to generate different soils through Porespy with polydisperse_spheres, which could generate various spheres as solid. Then, I plan to generate soil images through porosity and average pore diameter, and the pore diameter distribution can be controlled, maybe wide or narrow distribution.
Can you give me some advices on how to make it?
Thanks!

**Comment by jgostick** [comment]
The generators in porespy are not very sophisticated, and are only meant to provide fairly basic structures for testing and prototyping.  If you are serious about generating accurate images you'll probably need to come up with a recipe of your own.  [Here is an example](https://porespy.org/examples/generators/tutorials/Creating_Multiscale_Images.html) to give you an idea.

**Reply by morendong** [reply]
How about OpenPNM，which can generate a network with a porosity or pore diameter?

**Reply by jgostick** [reply]
If doing PNM simulations is the final goal then maybe, but I would myself try to generate images then do network extraction.  You do lots of a additional science and studies if you have the images.

For generating images you can look at geodict, which cost money but offers a lot more powerful generators

**Reply by morendong** [reply]
Yes! I first want to generate different images which represent soil with different pores. And the  extract pore network. Now  the soil images with different porosity can be generate through Porespy, but the pore diameter distribution are similar for different soil images, i.e., the pore distribution is narrow.
Maybe we can add some part to generate the soil image through some parameters, such as, porosity, pore average diameter, which would be more realistic and useful.
Thanks again for your kind reply and the Porespy!
